server.js

```diff
@@ -0,0 +1,386 @@
+        formDataObject["elementName"] === "index" ||
+        formDataObject["elementName"] === "404"
+      ) {
+        console.log("Invalid elementName");
+        request.on("end", function() {
+          response.writeHead(500, { "Content-Type": "application/json" });
+          let responseBody = {
+            error: "Invalid elementName"
+          };
+          response.end(JSON.stringify(responseBody));
+        });
+        // Create new HTML file and populate with POST data
+      } else {
+        let newElement = `<!DOCTYPE html>
+  <html lang="en">
+    <head>
+      <meta charset="UTF-8" />
+      <title>The Elements - ${formDataObject["elementName"]}</title>
+      <link rel="stylesheet" href="/css/styles.css" />
+    </head>
+    <body>
+      <h1>${formDataObject["elementName"]}</h1>
+      <h2>${formDataObject["elementSymbol"]}</h2>
+      <h3>Atomic number ${formDataObject["elementAtomicNumber"]}</h3>
+      <p>
+        ${formDataObject["elementDescription"]}
+      </p>
+      <p><a href="/">back</a></p>
+    </body>
+  </html>
+  `;
+        fs.writeFile(
+          "./public/" + formDataObject["elementName"].toLowerCase() + ".html",
+          newElement,
+          function(error) {
+            if (error) throw error;
+            console.log("File created successfully");
+          }
+        );
+        // Send success response
+        request.on("end", function() {
+          response.writeHead(200, { "Content-Type": "application/json" });
+          let responseBody = { success: "true" };
+          response.end(JSON.stringify(responseBody));
+        });
+      }
+    });
+  }
+
+  // DELETE REQUESTS
+
+  if (request.method === "DELETE") {
+    console.log(request.method + " " + request.url);
+    // Return 500 if file does not exist
+    if (!fs.existsSync("./public" + request.url)) {
+      console.log("resource " + request.url + " does not exist");
+      response.writeHead(500, { "Content-Type": "application/json" });
+      let responseBody = {
+        error: "resource " + request.url + " does not exist"
+      };
+      response.end(JSON.stringify(responseBody));
+      // Disallow deletion of essential files
+    } else if (
+      request.url === "/index.html" ||
+      request.url === "/404.html" ||
+      request.url === "/css/styles.css" ||
+      request.url === "/hydrogen.html" ||
+      request.url === "/helium.html"
+    ) {
+      console.log("resource " + request.url + " cannot be deleted");
+      response.writeHead(500, { "Content-Type": "application/json" });
+      let responseBody = {
+        error: "resource " + request.url + " cannot be deleted"
+      };
+      response.end(JSON.stringify(responseBody));
+    } else {
+      // Delete element html file
+      fs.unlinkSync("./public" + request.url);
+      console.log(request.url + " deleted successfully");
+      // Update index.html
+      fs.readFile("./public/index.html", function(error, data) {
+        if (error) throw error;
```

Comment:
Errors should be handled properly, they should send back a response with a appropriate status code and message.